cache/src/main/java/net/runelite/cache/MapImageDumper.java

```diff
@@ -580,53 +591,36 @@ else if (shape == 1)
 								}
 								else
 								{
+									shape--; //This is how jagex does it.
 									int drawX = xi * MAP_SCALE;
 									int drawY = (Region.Y - 1 - yi) * MAP_SCALE;
-									int[] tileShapes = TILE_SHAPE_2D[shape];
-									int[] tileRotations = TILE_ROTATION_2D[rotation];
+									rotation = convertTileRotation(rotation, shape);
+									shape = convertTileShape(shape);
 									if (underlayRgb != 0)
 									{
-										int rotIdx = 0;
-										for (int i = 0; i < Region.Z; ++i)
+										//blending between ground types
+										int idx = 0;
+										for (int iy = 0; iy < MAP_SCALE; ++iy)
 										{
-											int p1 = tileShapes[tileRotations[rotIdx++]] == 0 ? underlayRgb : overlayRgb;
-											int p2 = tileShapes[tileRotations[rotIdx++]] == 0 ? underlayRgb : overlayRgb;
-											int p3 = tileShapes[tileRotations[rotIdx++]] == 0 ? underlayRgb : overlayRgb;
-											int p4 = tileShapes[tileRotations[rotIdx++]] == 0 ? underlayRgb : overlayRgb;
-											pixels[drawX + 0][drawY + i] = p1;
-											pixels[drawX + 1][drawY + i] = p2;
-											pixels[drawX + 2][drawY + i] = p3;
-											pixels[drawX + 3][drawY + i] = p4;
+											for (int ix = 0; ix < MAP_SCALE; ++ix)
+											{
+												int p = TILE_SHAPE_2D[shape - 1][rotation][idx++] == 0 ? underlayRgb : overlayRgb;
+												setPixels(pixels, drawX + ix, drawY + iy, p);
+											}
 										}
 									}
 									else
 									{
+										//What is this?
```

Comment:
what is this?

Reply:
Just a note that I never figured out exactly what that code branch was used for. I've done some testing and its for drawing partial overlays, such as the brown on islands around tutorial island. I've updated the comment.